### PR TITLE
Restore the Pausetimes Apps in the UI

### PR DESCRIPTION
**Pull request by punchagan**
This PR re-enables the previously disabled Pausetimes apps.  The apps have been updated to display the data in the new benchmark directories `pausetimes_seq` and `pausetimes_par` produced using `olly latency`. 

The PR also: 
1.  Adds a script that fixes `pausetimes` bench files to be formatted similar to the new formatting implemented in https://github.com/tarides/runtime_events_tools/pull/13
2. Calls this script when copying data to the main branch to ensure any new data is formatted correctly. This is required until we are able to use the latest version of `runtime_events_tools` in Sandmark, which depends on `ppxlib` being compatible with OCaml 5.1 and 5.2. 
3. Updates existing bench files `pausetimes_*` to use the new formatting, along with fixing an issue with the bench files having pretty printed JSON which breaks the assumption that every line in the bench file is a valid JSON string. This was fixed in https://github.com/ocaml-bench/sandmark/pull/452

### CODE REVIEW

File: app/apps/instrumented_pausetimes_parallel.py [hunk 1]
```diff
@@ -1,27 +1,21 @@
-import streamlit as st
```
Comment (kayceesrk) [inline app/apps/instrumented_pausetimes_parallel.py:1]: The runtime is no longer instrumented. Best to call it `pausetimes_parallel.py`. Similarly for the sequential one.

**Comment by kayceesrk** [comment]
Left a minor comment. Otherwise, LGTM.

FWIW, I'm not able to review this UI/UX PRs easily without being able to test the app. We'll have to merge the PR and then fix any issues observed. 

Do we have a staging area where we can deploy the app and then test it before we deploy it in production?

**Review by kayceesrk — APPROVED** [review]

**Comment by punchagan** [comment]
@kayceesrk Thanks for the review!

> Do we have a staging area where we can deploy the app and then test it before we deploy it in production?

We don't have a staging area, but I just deployed the branch on the Streamlit (free) hosting. You can check it out [here](https://punchagan-sandmark-nightly-appapp-restore-pausetimes-2s4otk.streamlit.app/).  

The deployment doesn't seem to use the exact versions of dependencies specified in our `requirements.txt`, probably since they try to be smart by installing some dependencies (`pandas`, `altair`, etc.) by default for all the apps. Currently the Parallel pausetimes app is crashing, but the sequential one works. I'm looking around to see if there's a quick fix for the crash.  

> Left a minor comment. Otherwise, LGTM.

Thanks, I'll address this soon, along with any other comments you may have after using the UI.

**Comment by punchagan** [comment]
> I'm looking around to see if there's a quick fix for the crash.

We were using an older version of `streamlit` in our `requirements.txt` and the deployment logic for supporting older versions of Streamlit on the cloud was causing issues with other dependencies. I've pushed a commit to update our dependency to the latest version of Streamlit, and everything seems to work okay. 

@kayceesrk You can look at the deployed branch [here](https://punchagan-sandmark-nightly-appapp-restore-pausetimes-2s4otk.streamlit.app/?app=Instrumented+Pausetimes+Parallel&pausetimes_par_00=turing&pausetimes_par_num_variants=2&pausetimes_par_11=20230529&pausetimes_par_10=navajo&pausetimes_par_01=20230530&pausetimes_par_find_by=variant&pausetimes_par_02=turing_5.2.0%2Btrunk%2Bbartoszmodelski%2Bpr12212%2Bpausetimes_par_20230530_a6f309f&pausetimes_par_12=navajo_5.2.0%2Btrunk%2Bbartoszmodelski%2Bpr12212%2Bpausetimes_par_20230529_a6f309f) for a more complete review. Thanks!

**Comment by kayceesrk** [comment]
image

The features look really awesome.

1. Remove "Instrumented" from all the locations.
2. Having both "sequential benchmarks" and "pausetimes sequential" seems odd. Can the "pausetimes sequential" appear at the bottom of the "sequential benchmarks" results? 
3. In fact, the pausetimes are no longer computed on instrumented programs. If we are clever about isolating noise, the pausetimes numbers could be computed along with the sequential and parallel runs. FWIW earlier pausetimes were computed on the instrumented runtime, which caused slowdown of programs. This is no longer the case as the pausetimes are read from the shared ring buffer. This will cut down the running time of the nightly runs by 1/2 CC @Sudha247 thoughts?

Please do (1) in this PR as it is quick. For (2), given that the results are computed from different runs unlike suggested in (3), it might be a little tricky. As a stop-gap measure, may I suggest reordering the pages in the following way?

```
Goto
* Home
* Sequential -- throughput
* Sequential -- latency
* Parallel -- throughput
* Parallel -- latency
```

It may be useful to track (3) in a separate issue.

**Comment by Sudha247** [comment]
I also think we can extract both running times and pausetimes from the same run. I've observed the overheads added by `olly` to be quite low, though we may have to confirm this for the benchmarks in sandmark. Good idea to track this in a separate issue.

**Comment by punchagan** [comment]
> Good idea to track this in a separate issue.

I've created https://github.com/ocaml-bench/sandmark/issues/454. Feel free to edit or add more information.

**Comment by kayceesrk** [comment]
LGTM. Thanks.

**Pull request merged**